File: GPIO-Latency/measurements/reply.c
#include <wiringPi.h>
#include <sched.h>
#define GPIO_IN 4 //bcm 23
#define GPIO_OUT 5 //bcm 24

int main(void)
{
    const struct sched_param priority = {80};
    sched_setscheduler(0, SCHED_RR, &priority);
    //SCHED_OTHER, SCHED_BATCH, SCHED_FIFO, SCHED_RR

    wiringPiSetup() ;
    pinMode(GPIO_OUT, OUTPUT);
    pinMode(GPIO_IN, INPUT);
    while(1){
        while(digitalRead(GPIO_IN))
            ;
        digitalWrite(GPIO_OUT, LOW);
        digitalWrite(GPIO_OUT, HIGH);
    }
    return 0;
}

File: GPIO-Latency/measurements/reply.py
#!/usr/bin/env python3
#apt install python3-rpi.gpio

import os
import RPi.GPIO as GPIO

GPIO_IN = 23
GPIO_OUT = 24

if __name__ == '__main__':
    param = os.sched_param(os.sched_get_priority_max(os.SCHED_FIFO))
    os.sched_setscheduler(80, os.SCHED_FIFO, param)

    GPIO.setmode(GPIO.BCM)
    #GPIO.setwarnings(False)

    # Set up the GPIO channels - one input and one output
    GPIO.setup(GPIO_IN, GPIO.IN)
    GPIO.setup(GPIO_OUT, GPIO.OUT)

    GPIO.output(GPIO_OUT, GPIO.HIGH)
    while True:
        try:
            while(GPIO.input(GPIO_IN)):
                pass
            GPIO.output(GPIO_OUT, GPIO.LOW)
            GPIO.output(GPIO_OUT, GPIO.HIGH)
        except KeyboardInterrupt:
            print('\nBye')
            GPIO.cleanup()
            exit(0)
